### PR TITLE
[AIC-py][eval] refactor enough to mock aiconfig and add more testing

### PR DESCRIPTION
**Pull request by jonathanlastmileai**
[AIC-py][eval] refactor enough to mock aiconfig and add more testing

---
Stack created with [Sapling](https://sapling-scm.com). Best reviewed with [ReviewStack](https://reviewstack.dev/lastmile-ai/aiconfig/pull/430).
* #432
* __->__ #430
* #428
* #426

### CODE REVIEW

File: python/tests/test_eval.py [hunk 1]
```diff
@@ -23,13 +53,17 @@ def test_metrics():
 
 @pytest.mark.asyncio
 async def test_run_with_inputs_sanity_check():
-    """No easy way to mock LLM calls,
-    so just test the imports and sanity check output."""
+    """No easy way to mock LLM calls from outside run_test_suite_with_inputs.
+
+    Instead, give empty list and just test the imports and sanity check output."""
```
Comment (rossdanlm) [inline python/tests/test_eval.py:58]: Give an actual example on how to call this (or do GH link example to code that does this)

File: python/tests/test_eval.py [hunk 2]
```diff
@@ -23,13 +53,17 @@ def test_metrics():
...
+
+    path = os.path.join(
+        current_dir(),
+        "../src/aiconfig/eval/examples/travel/travel_parametrized.aiconfig.json",
```
Comment (rossdanlm) [inline python/tests/test_eval.py:62]: Use the `path` vary instead of `../src`

File: python/tests/test_eval.py [hunk 3]
```diff
@@ -1,17 +1,47 @@
...
 from hypothesis import given
 
+from aiconfig.model_parser import InferenceOptions
+
+
+def current_dir():
+    return os.path.dirname(os.path.realpath(__file__))
+
+
+class MockAIConfigRuntime(AIConfigRuntime):
```
Comment (rossdanlm) [inline python/tests/test_eval.py:27]: Let's make this separate file to keep things cleaner

**Review by rossdanlm — APPROVED** [review]
accepting to unblock

File: python/tests/test_eval.py [hunk 4]
```diff
@@ -1,17 +1,47 @@
...
+        params: dict[Any, Any] | None = None,
+        options: InferenceOptions | None = None,
+        **kwargs,
+    ) -> str:
+        params_ = params or {}
+        assert params_.keys() == {
+            "the_query"
+        }, 'For eval, AIConfig params must have just the key "the_query".'
+        the_query = params_["the_query"]
+        return f"output_for_{prompt_name}_the_query_{the_query}"
```
[the remark this answers is not in the document]
Comment (jonathanlastmileai) [reply python/tests/test_eval.py:43]: This is not for display, it's the mock return value.

File: python/tests/test_eval.py (hunk 1 above)
Comment (jonathanlastmileai) [reply python/tests/test_eval.py:58]: I'm confused what the ask was here @rossdanlm

**Pull request merged**